doc/reference/kernel/threads/workqueue.rst

```diff
@@ -287,17 +336,185 @@ by calling :c:func:`k_delayed_work_cancel`.
 
 .. warning::
    All of these operations can fail as described in :ref:`k_delayed_work`.
+   The replacement API in :ref:`k_delayable_work` should be used instead.
+
+Submitting a Delayable Work Item
+================================
+
+A delayable work item is defined using a variable of type
+:c:struct:`k_work_delayable`. It must then be initialized by calling
+:c:func:`k_work_delayable_init`.
+
+An initialized delayable work item can be scheduled for submission to the
+system workqueue by calling :c:func:`k_work_schedule`, or to a specified
+workqueue by calling :c:func:`k_work_schedule_for_queue`.  A delayable work
+item that has been scheduled but has not yet been submitted can be cancelled
+by calling :c:func:`k_work_cancel`.
 
 Suggested Uses
 **************
 
-Use the system workqueue to defer complex interrupt-related processing
-from an ISR to a cooperative thread. This allows the interrupt-related
-processing to be done promptly without compromising the system's ability
-to respond to subsequent interrupts, and does not require the application
-to define an additional thread to do the processing.
+Use the system workqueue to defer complex interrupt-related processing from an
+ISR to a cooperative thread. This allows the interrupt-related processing to
+be done promptly without compromising the system's ability to respond to
+subsequent interrupts, and does not require the application to define an
+additional thread to do the processing.
+
+Simple ISR Offloading
+=====================
```

Comment:
Seems like this example, which isn't related to the delayable work API, should be in a separate patch?

Reply:
Done.